Building for GNU/Linux with GLES and no X11, possible?

Hi @ptitSeb 

Is it possible to build this for GNU/Linux with GLES only?
I saw that, when building for Pandora (`-DPANDORA=ON`) libglew is still needed, which I understand is desktop-GL only.

I also tried `-DGLES=ON`, but got linking errors:
```
[ 65%] Linking CXX executable Eldritch
cd /home/manuel/src/Eldritch/Code/b4/Projects/Eldritch && /usr/bin/cmake -E cmake_link_script CMakeFiles/Eldritch.dir/link.txt --verbose=1
/usr/bin/c++    -rdynamic CMakeFiles/Eldritch.dir/src/Actions/wbactioneldaddblock.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldbanktransaction.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldbindinput.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldcheckcone.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldcheckline.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldchecksphere.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactionelddamageworld.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldgiveitem.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldgotolevel.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldlaunchwebsite.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldplayanim.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldplayhandanim.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldremoveblock.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldsetlight.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldsetpersistentvar.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldshowbook.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldspawnentity.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldstopmotion.cpp.o CMakeFiles/Eldritch.dir/src/Actions/wbactioneldtweetrip.cpp.o CMakeFiles/Eldritch.dir/src/AnimEvents/animeventeldexecuteaction.cpp.o CMakeFiles/Eldritch.dir/src/AnimEvents/animeventeldsendevent.cpp.o CMakeFiles/Eldritch.dir/src/BTNodes/rodinbtnodeeldlookat.cpp.o CMakeFiles/Eldritch.dir/src/BTNodes/rodinbtnodeeldmoveto.cpp.o CMakeFiles/Eldritch.dir/src/BTNodes/rodinbtnodeeldplayanim.cpp.o CMakeFiles/Eldritch.dir/src/BTNodes/rodinbtnodeeldplaybark.cpp.o CMakeFiles/Eldritch.dir/src/BTNodes/rodinbtnodeeldstopmoving.cpp.o CMakeFiles/Eldritch.dir/src/BTNodes/rodinbtnodeeldturntoward.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldaimotion.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldammobag.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldanchor.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldcamera.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldcharacter.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldclimbable.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldcollision.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompelddamager.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompelddoor.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompelddropshadow.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldendgamecounter.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldfaction.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldfootsteps.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldfrobbable.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldfrobber.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldhands.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldhard.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldhardlistener.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldheadtracker.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldhealth.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldhudmarker.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldicicles.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldinventory.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompelditem.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldkeyring.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldlight.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldlock.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldmapmarker.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldmarkup.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldmesh.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldminimap.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldparticles.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldpickup.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldplayer.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldpowerteleport.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldrespawner.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldrope.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsensor.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsensordamage.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsensorhearing.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsensormarkup.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsensorpoll.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsensortheft.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsensorvision.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsleeper.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldsound.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldspikes.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldthinker.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldthinkerhearwormtarget.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldthinkerpatrol.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldthinkertarget.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldtransform.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldtrapbolt.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldusable.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldvisible.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldwallet.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldwatson.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbcompeldweapon.cpp.o CMakeFiles/Eldritch.dir/src/Components/wbeldritchcomponent.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeelddistance.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldgetcharactervo.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldgetfaction.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldgetitem.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldgetlocation.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldgetpersistentvar.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldgetslot.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldgetvelocity.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldhardscalar.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldhardswitch.cpp.o CMakeFiles/Eldritch.dir/src/PEs/wbpeeldplayer.cpp.o CMakeFiles/Eldritch.dir/src/SDPs/sdpelddropshadow.cpp.o CMakeFiles/Eldritch.dir/src/SDPs/sdpeldhud.cpp.o CMakeFiles/Eldritch.dir/src/SDPs/sdpeldhudcalibration.cpp.o CMakeFiles/Eldritch.dir/src/SDPs/sdpeldlit.cpp.o CMakeFiles/Eldritch.dir/src/SDPs/sdpeldpost.cpp.o CMakeFiles/Eldritch.dir/src/SDPs/sdpeldworld.cpp.o CMakeFiles/Eldritch.dir/src/Screens/uiscreen-eldbindinputs.cpp.o CMakeFiles/Eldritch.dir/src/Screens/uiscreen-eldcredits.cpp.o CMakeFiles/Eldritch.dir/src/Screens/uiscreen-eldmirror.cpp.o CMakeFiles/Eldritch.dir/src/Screens/uiscreen-eldsetres.cpp.o CMakeFiles/Eldritch.dir/src/eldritchbank.cpp.o CMakeFiles/Eldritch.dir/src/eldritchfactions.cpp.o CMakeFiles/Eldritch.dir/src/eldritchframework.cpp.o CMakeFiles/Eldritch.dir/src/eldritchgame.cpp.o CMakeFiles/Eldritch.dir/src/eldritchirradiance.cpp.o CMakeFiles/Eldritch.dir/src/eldritchmesh.cpp.o CMakeFiles/Eldritch.dir/src/eldritchmusic.cpp.o CMakeFiles/Eldritch.dir/src/eldritchnav.cpp.o CMakeFiles/Eldritch.dir/src/eldritchparticles.cpp.o CMakeFiles/Eldritch.dir/src/eldritchpersistence.cpp.o CMakeFiles/Eldritch.dir/src/eldritchroom.cpp.o CMakeFiles/Eldritch.dir/src/eldritchsaveload.cpp.o CMakeFiles/Eldritch.dir/src/eldritchsound3dlistener.cpp.o CMakeFiles/Eldritch.dir/src/eldritchtargetmanager.cpp.o CMakeFiles/Eldritch.dir/src/eldritchtools.cpp.o CMakeFiles/Eldritch.dir/src/eldritchwbscenehelpers.cpp.o CMakeFiles/Eldritch.dir/src/eldritchworld.cpp.o CMakeFiles/Eldritch.dir/src/eldritchworldgen.cpp.o CMakeFiles/Eldritch.dir/src/main.cpp.o CMakeFiles/Eldritch.dir/src/uiinputmapeldritch.cpp.o  -o Eldritch   -L/home/manuel/src/Eldritch/Code/Libraries/Audio  -L/home/manuel/src/Eldritch/Code/Libraries/Audio/lib  -L/home/manuel/src/Eldritch/Code/Libraries/Math  -L/home/manuel/src/Eldritch/Code/Libraries/Framework  -L/home/manuel/src/Eldritch/Code/Libraries/UI  -L/home/manuel/src/Eldritch/Code/Libraries/2D  -L/home/manuel/src/Eldritch/Code/Libraries/3D  -L/home/manuel/src/Eldritch/Code/Libraries/Core  -L/home/manuel/src/Eldritch/Code/Libraries/Rodin  -L/home/manuel/src/Eldritch/Code/Libraries/Workbench  -L/usr/local/lib  -Wl,-rpath,/home/manuel/src/Eldritch/Code/Libraries/Audio:/home/manuel/src/Eldritch/Code/Libraries/Audio/lib:/home/manuel/src/Eldritch/Code/Libraries/Math:/home/manuel/src/Eldritch/Code/Libraries/Framework:/home/manuel/src/Eldritch/Code/Libraries/UI:/home/manuel/src/Eldritch/Code/Libraries/2D:/home/manuel/src/Eldritch/Code/Libraries/3D:/home/manuel/src/Eldritch/Code/Libraries/Core:/home/manuel/src/Eldritch/Code/Libraries/Rodin:/home/manuel/src/Eldritch/Code/Libraries/Workbench:/usr/local/lib ../../Libraries/Core/libCore.a ../../Libraries/Math/libMath.a ../../Libraries/2D/lib2D.a ../../Libraries/3D/lib3D.a ../../Libraries/Audio/libAudio.a ../../Libraries/Workbench/libWorkbench.a ../../Libraries/Rodin/libRodin.a ../../Libraries/UI/libUI.a ../../Libraries/Framework/libFramework.a -lSDL2 -lGLESv2 -lopenal -lalut -lvorbisfile ../../Libraries/2D/lib2D.a ../../Libraries/3D/lib3D.a ../../Libraries/Core/libCore.a -lz ../../Libraries/Math/libMath.a ../../Libraries/UI/libUI.a -lopenal 
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::Initialize()':
gl2renderer.cpp:(.text+0x2a1): undefined reference to `glFrontFace'
/usr/bin/ld: gl2renderer.cpp:(.text+0x2b8): undefined reference to `glGetIntegerv'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::RenderBucket(Bucket*, ViewportPass const*)':
gl2renderer.cpp:(.text+0xc31): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xc76): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xc97): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xcdc): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xcfd): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xd42): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xd68): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xdad): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xdd3): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xe18): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xe3b): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xe80): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xea6): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xeeb): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xf11): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xf56): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xf77): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0xfbc): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0xfdd): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1022): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1043): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1088): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0x10a9): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0x10ee): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1111): undefined reference to `glEnableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1156): undefined reference to `glVertexAttribPointer'
/usr/bin/ld: gl2renderer.cpp:(.text+0x118b): undefined reference to `glDisableVertexAttribArray'
/usr/bin/ld: gl2renderer.cpp:(.text+0x124a): undefined reference to `glDrawElements'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::Clear(unsigned int, unsigned int, float, unsigned int)':
gl2renderer.cpp:(.text+0x1359): undefined reference to `glClearColor'
/usr/bin/ld: gl2renderer.cpp:(.text+0x137a): undefined reference to `glClearDepthf'
/usr/bin/ld: gl2renderer.cpp:(.text+0x13a8): undefined reference to `glClearStencil'
/usr/bin/ld: gl2renderer.cpp:(.text+0x13b8): undefined reference to `glDepthMask'
/usr/bin/ld: gl2renderer.cpp:(.text+0x13c2): undefined reference to `glClear'
/usr/bin/ld: gl2renderer.cpp:(.text+0x13d2): undefined reference to `glDepthMask'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetRenderTarget(IRenderTarget*)':
gl2renderer.cpp:(.text+0x18c0): undefined reference to `glBindFramebuffer'
/usr/bin/ld: gl2renderer.cpp:(.text+0x19c3): undefined reference to `glViewport'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetShaderProgram(IShaderProgram*)':
gl2renderer.cpp:(.text+0x1a94): undefined reference to `glUseProgram'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::glSetEnabled(unsigned int, bool)':
gl2renderer.cpp:(.text+0x1af0): undefined reference to `glEnable'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1afc): undefined reference to `glDisable'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetCullMode(ECullMode)':
gl2renderer.cpp:(.text+0x1b3f): undefined reference to `glDisable'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1b4b): undefined reference to `glEnable'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1b69): undefined reference to `glCullFace'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetZWriteEnable(EZWriteEnable)':
gl2renderer.cpp:(.text+0x1c06): undefined reference to `glDepthMask'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetBlend(EBlend, EBlend)':
gl2renderer.cpp:(.text+0x1ce1): undefined reference to `glBlendFunc'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetTexture(unsigned int, ITexture*)':
gl2renderer.cpp:(.text+0x1d0c): undefined reference to `glActiveTexture'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1d68): undefined reference to `glBindTexture'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetAddressing(unsigned int, ETextureAddress, ETextureAddress)':
gl2renderer.cpp:(.text+0x1e0f): undefined reference to `glTexParameteri'
/usr/bin/ld: gl2renderer.cpp:(.text+0x1e37): undefined reference to `glTexParameteri'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetMinMipFilters(unsigned int, ETextureFilter, ETextureFilter)':
gl2renderer.cpp:(.text+0x1ef7): undefined reference to `glTexParameteri'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetMagFilter(unsigned int, ETextureFilter)':
gl2renderer.cpp:(.text+0x1f8a): undefined reference to `glTexParameteri'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetPixelShaderFloat4(HashedString const&, float const*, unsigned int)':
gl2renderer.cpp:(.text+0x213c): undefined reference to `glUniform4fv'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::SetPixelShaderMatrix(HashedString const&, float const*, unsigned int)':
gl2renderer.cpp:(.text+0x2328): undefined reference to `glUniformMatrix4fv'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2rendertarget.cpp.o): in function `GL2RenderTarget::~GL2RenderTarget()':
gl2rendertarget.cpp:(.text+0x1d5): undefined reference to `glDeleteRenderbuffers'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x1f5): undefined reference to `glDeleteRenderbuffers'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x215): undefined reference to `glDeleteFramebuffers'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2rendertarget.cpp.o): in function `GL2RenderTarget::Initialize(SRenderTargetParams const&)':
gl2rendertarget.cpp:(.text+0x317): undefined reference to `glGenFramebuffers'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x3a6): undefined reference to `glGenTextures'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x423): undefined reference to `glBindTexture'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x462): undefined reference to `glTexImage2D'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x47a): undefined reference to `glTexParameteri'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x48e): undefined reference to `glTexParameteri'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x4a2): undefined reference to `glTexParameteri'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x4b6): undefined reference to `glTexParameteri'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x4eb): undefined reference to `glBindTexture'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x510): undefined reference to `glGenRenderbuffers'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x58d): undefined reference to `glBindRenderbuffer'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x5b0): undefined reference to `glRenderbufferStorage'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x5c5): undefined reference to `glGenRenderbuffers'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x5d8): undefined reference to `glBindRenderbuffer'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x5fb): undefined reference to `glRenderbufferStorage'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x60a): undefined reference to `glBindRenderbuffer'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x61d): undefined reference to `glBindFramebuffer'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x640): undefined reference to `glFramebufferTexture2D'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x65d): undefined reference to `glFramebufferRenderbuffer'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x67a): undefined reference to `glFramebufferRenderbuffer'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x684): undefined reference to `glCheckFramebufferStatus'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x6fe): undefined reference to `glBindFramebuffer'
/usr/bin/ld: gl2rendertarget.cpp:(.text+0x711): undefined reference to `glBindFramebuffer'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2shaderprogram.cpp.o): in function `GL2ShaderProgram::~GL2ShaderProgram()':
gl2shaderprogram.cpp:(.text+0x225): undefined reference to `glDetachShader'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x2c1): undefined reference to `glDetachShader'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x2cf): undefined reference to `glDeleteProgram'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2shaderprogram.cpp.o): in function `GL2ShaderProgram::Initialize(IVertexShader*, IPixelShader*, IVertexDeclaration*)':
gl2shaderprogram.cpp:(.text+0x4c6): undefined reference to `glCreateProgram'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x5e2): undefined reference to `glAttachShader'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x687): undefined reference to `glAttachShader'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x6b1): undefined reference to `glLinkProgram'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x6cb): undefined reference to `glGetProgramiv'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x6f1): undefined reference to `glGetProgramiv'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x740): undefined reference to `glGetProgramInfoLog'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2shaderprogram.cpp.o): in function `GL2ShaderProgram::BindAttributes(IVertexDeclaration*) const':
gl2shaderprogram.cpp:(.text+0x8bd): undefined reference to `glBindAttribLocation'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x8e5): undefined reference to `glBindAttribLocation'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x90d): undefined reference to `glBindAttribLocation'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x93a): undefined reference to `glBindAttribLocation'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0x967): undefined reference to `glBindAttribLocation'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2shaderprogram.cpp.o):gl2shaderprogram.cpp:(.text+0x991): more undefined references to `glBindAttribLocation' follow
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2shaderprogram.cpp.o): in function `GL2ShaderProgram::BuildUniformTable()':
gl2shaderprogram.cpp:(.text+0xaff): undefined reference to `glGetProgramiv'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0xb79): undefined reference to `glGetActiveUniform'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0xb98): undefined reference to `glGetUniformLocation'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2shaderprogram.cpp.o): in function `GL2ShaderProgram::SetSamplerUniforms()':
gl2shaderprogram.cpp:(.text+0xd60): undefined reference to `glUseProgram'
/usr/bin/ld: gl2shaderprogram.cpp:(.text+0xdd5): undefined reference to `glUniform1i'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2texture.cpp.o): in function `GL2Texture::~GL2Texture()':
gl2texture.cpp:(.text+0xcc): undefined reference to `glDeleteTextures'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2texture.cpp.o): in function `GL2Texture::CreateTexture(unsigned char*)':
gl2texture.cpp:(.text+0x1ff): undefined reference to `glGenTextures'
/usr/bin/ld: gl2texture.cpp:(.text+0x27c): undefined reference to `glBindTexture'
/usr/bin/ld: gl2texture.cpp:(.text+0x382): undefined reference to `glTexImage2D'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2texture.cpp.o): in function `GL2Texture::CreateTextureFromDDS(IDataStream const&)':
gl2texture.cpp:(.text+0x87b): undefined reference to `glGenTextures'
/usr/bin/ld: gl2texture.cpp:(.text+0x907): undefined reference to `glBindTexture'
/usr/bin/ld: gl2texture.cpp:(.text+0xacd): undefined reference to `glTexImage2D'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2vertexshader.cpp.o): in function `GL2VertexShader::~GL2VertexShader()':
gl2vertexshader.cpp:(.text+0x4db): undefined reference to `glDeleteShader'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2vertexshader.cpp.o): in function `GL2VertexShader::Initialize(IDataStream const&)':
gl2vertexshader.cpp:(.text+0x605): undefined reference to `glCreateShader'
/usr/bin/ld: gl2vertexshader.cpp:(.text+0x6af): undefined reference to `glShaderSource'
/usr/bin/ld: gl2vertexshader.cpp:(.text+0x6c0): undefined reference to `glCompileShader'
/usr/bin/ld: gl2vertexshader.cpp:(.text+0x6da): undefined reference to `glGetShaderiv'
/usr/bin/ld: gl2vertexshader.cpp:(.text+0x700): undefined reference to `glGetShaderiv'
/usr/bin/ld: gl2vertexshader.cpp:(.text+0x74f): undefined reference to `glGetShaderInfoLog'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2pixelshader.cpp.o): in function `GL2PixelShader::~GL2PixelShader()':
gl2pixelshader.cpp:(.text+0x6b): undefined reference to `glDeleteShader'
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2pixelshader.cpp.o): in function `GL2PixelShader::Initialize(IDataStream const&)':
gl2pixelshader.cpp:(.text+0x14c): undefined reference to `glCreateShader'
/usr/bin/ld: gl2pixelshader.cpp:(.text+0x23f): undefined reference to `glShaderSource'
/usr/bin/ld: gl2pixelshader.cpp:(.text+0x250): undefined reference to `glCompileShader'
/usr/bin/ld: gl2pixelshader.cpp:(.text+0x26a): undefined reference to `glGetShaderiv'
/usr/bin/ld: gl2pixelshader.cpp:(.text+0x290): undefined reference to `glGetShaderiv'
/usr/bin/ld: gl2pixelshader.cpp:(.text+0x2df): undefined reference to `glGetShaderInfoLog'
collect2: error: ld returned 1 exit status
make[2]: *** [Projects/Eldritch/CMakeFiles/Eldritch.dir/build.make:1988: Projects/Eldritch/Eldritch] Error 1
make[2]: Leaving directory '/home/manuel/src/Eldritch/Code/b4'
make[1]: *** [CMakeFiles/Makefile2:623: Projects/Eldritch/CMakeFiles/Eldritch.dir/all] Error 2
make[1]: Leaving directory '/home/manuel/src/Eldritch/Code/b4'
make: *** [Makefile:84: all] Error 2
```

@ptitSeb The problem seems to be that `Libraries/3D/CMakeFiles/3D.dir/link.txt` is missing -lGLESv2, etc. If we look closely at the linking errors:

```
/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::Initialize()':
gl2renderer.cpp:(.text+0x2a1): undefined reference to `glFrontFace'
```

So it's lib3D.a linkage that's failing.

@ptitSeb 

If you look at the first error line I get:
```

/usr/bin/ld: ../../Libraries/3D/lib3D.a(gl2renderer.cpp.o): in function `GL2Renderer::Initialize()':
gl2renderer.cpp:(.text+0x2a1): undefined reference to `glFrontFace'


```

...you will see that the missing reference is in `lib3D.a`
But the problem is happening on the main Eldritch executable linkage.

If I look at the linking line, I can see this:
`
../../Libraries/Workbench/libWorkbench.a ../../Libraries/Rodin/libRodin.a ../../Libraries/UI/libUI.a ../../Libraries/Framework/libFramework.a -lSDL2 -lGLESv2 -lopenal -lalut -lvorbisfile ../../Libraries/2D/lib2D.a ../../Libraries/3D/lib3D.a ../../Libraries/Core/libCore.a -lz ../../Libraries/Math/libMath.a ../../Libraries/UI/libUI.a -lopenal`

As you can see, `lib3D.a` is passed AFTER the `-lGLESv2` flag!
I believe that's the problem. But how can I modify that?

PS: I have SDL2 with desktop GL and GLES support, of course. The `testgles2` included in the SDL2 sources runs here nicely.

--- a/Code/Libraries/3D/CMakeLists.txt
+++ b/Code/Libraries/3D/CMakeLists.txt
@@ -71,7 +71,7 @@ endif(GLES)
 add_library(3D ${3d_src})
 
 if(GLES)
-	target_link_libraries(3D Core Math)
+	target_link_libraries(3D Core Math GLESv2)
 else(GLES)
 	target_link_libraries(3D Core Math GL GLEW)
 endif(GLES)

--- a/Code/Libraries/Audio/CMakeLists.txt
+++ b/Code/Libraries/Audio/CMakeLists.txt
@@ -19,6 +19,4 @@ set(audio_src
 )
 
 add_library(Audio ${audio_src})
-target_link_libraries(Audio
-	SoLoud
-)
+target_link_libraries(Audio SoLoud)

--- a/Code/Projects/Eld/CMakeLists.txt
+++ b/Code/Projects/Eld/CMakeLists.txt
@@ -21,7 +21,6 @@ link_directories(${CMAKE_SOURCE_DIR}/Libraries/3D/)
 link_directories(${CMAKE_SOURCE_DIR}/Libraries/Core/)
 link_directories(${CMAKE_SOURCE_DIR}/Libraries/Rodin/)
 link_directories(${CMAKE_SOURCE_DIR}/Libraries/Workbench/)
-link_directories(/usr/local/lib)
 
 set(eld_src
 	src/Achievements/achievementmanager_steam.cpp
@@ -173,7 +172,7 @@ set(eld_src
 add_executable(Eld ${eld_src})
 
 if(GLES)
-	target_link_libraries(Eld Core Math 3D Audio Workbench Rodin UI Framework SDL2 GLESv2 openal)
+	target_link_libraries(Eld Math Workbench Core 3D Audio Rodin UI Framework SDL2 openal)
 else(GLES)
-	target_link_libraries(Eld Core Math 3D Audio Workbench Rodin UI Framework SDL2 GL GLEW openal)
+	target_link_libraries(Eld Math Workbench Core 3D Audio Rodin UI Framework SDL2 openal)
 endif(GLES)

